frontend/src/pages/chat/components/chat-layout/chat-layout.tsx

```diff
@@ -70,20 +75,27 @@ const ChatLayout: React.FC<Properties> = ({ filter }) => {
       behavior: 'smooth',
       block: 'end',
     });
-  }, [dispatch, currentChatMessages.length]);
+  }, [dispatch, currentChatMessagesLength]);
 
   return (
     <>
       <ChatHeader />
       <div className={styles['chat-body']}>
         {hasId &&
-          currentChatMessages.map((item) => {
+          Object.entries(currentChatMessages).map(([date, group]) => {
             return (
-              <ChatMessage
-                key={item.id}
-                message={item.message}
-                isSender={item.senderId === authenticatedUser.id}
-              />
+              <React.Fragment key={date}>
+                <ChatDivider date={new Date(date)} />
```

Comment:
DIvider even for a first item?

Reply:
but we need divider for the first item 🤔